articles/contributing/style-guide.md

```diff
@@ -0,0 +1,451 @@
+---
+title: Q# Style Guide | Microsoft Docs
+description: Q# Style Guide
+author: cgranade
+ms.author: chgranad
+ms.date: 10/12/2018
+ms.topic: article
+uid: microsoft.quantum.contributing.style
+---
+
+# Q# Style Guide #
+## General Conventions ##
+
+The conventions listed here are suggestions only, and should likely be disregarded when they would result in less readable or useful code.
+
+# [Guidance](#tab/guidance) 
+
+We suggest:
+
+- Never disregard a convention unless you’re doing so intentionally in order to provide more readable and understandable code for your users.
+
+# [Examples](#tab/examples)
+
+***
+
+## Naming Conventions ##
+
+In offering the Quantum Development Kit, we strive for function and operation names that help quantum developers write programs that are easy to read and that minimize surprise.
+An important part of that is that when we choose names for functions, operations, and types, we are establishing the *vocabulary* that programmers use to express quantum concepts; with our choices, we either help or hinder them in their effort to clearly communicate.
+This places a responsibility on us to make sure that the names we introduce lend themselves to clarity rather than obscurity.
+In this section, we detail how we meet this obligation in terms of explicit guidance that helps us do the best by the Q# development community.
+
+### Operations and Functions ###
+
+The first thing that a name should establish is whether a given symbol represents a function or an operation.
+The difference between functions and operations is critical to understanding how a block of code behaves.
+To communicate this distinction to programmers, we refer back to that in Q#, quantum computation proceeds through the effects of operations.
```

Comment:
awkward sentence.

Reply:
Thanks for the catch! I took a stab at rewording it, please let me know what you think.

Reply:
I didn't see a change, so I edited.  Do adjust if needed.